### ISSUE TITLE
Dimension ending with "u" not shown correctly in print(Dataset)

### ISSUE THREAD

**Issue by markusReinert** [Unidata/netcdf4-python#983]
* Version: 1.4.1 (but the line which most likely produces the error has not changed since then, see below)
* Python 3.6.7 on Ubuntu 18.04

A dimension whose name ends with the letter "u" is not shown correctly if print is invoked on a Dataset. However, it is shown correctly if print is invoked on the variables of the Dataset.

In fluid mechanics, one uses staggered grids, that means different quantities are defined at different coordinates. One then uses notations like "x_u", "y_u", "z_u" to denote the coordinates of the quantity "u" (velocity). If I open a netCDF file with these dimensions as a Dataset and call print on this Dataset, the dimensions are shown as "x_", "y_", "z_" with the "u" missing.

Minimal working example:
```
import netCDF4 as nc
d = nc.Dataset("tmp.nc", "w")
d.createDimension("x_u")
d.createVariable("u", float, ("x_u",))

# Wrong behaviour:
print(d)
# Result:
#<class 'netCDF4._netCDF4.Dataset'>
#root group (NETCDF4 data model, file format HDF5):
#    dimensions(sizes): x_u(0)
#    variables(dimensions): float64 u(x_)
#    groups:

# Correct behaviour:
print(d.variables)
```

The problem is most likely due to line 2428 in file netCDF4/_netCDF4.pyx, the function `__unicode__` of the Dataset class, which has the statement: `.replace("u'",""))\`.

**Comment by jswhit** [comment]
Fix is in pull request #984

**Comment by jswhit** [comment]
PR #984 merged
